src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCITriggerService.java

```diff
@@ -273,4 +273,17 @@ private BuildConfig getBuildConfig(ProgrammingExerciseParticipation participatio
         return new BuildConfig(buildScript, dockerImage, commitHashToBuild, assignmentCommitHash, testCommitHash, branch, programmingLanguage, projectType,
                 staticCodeAnalysisEnabled, sequentialTestRunsEnabled, testwiseCoverageEnabled, resultPaths);
     }
+
+    private int determinePriority(ProgrammingExercise programmingExercise) {
+        if (programmingExercise.isExamExercise()) {
+            boolean isTestExam = programmingExercise.getExerciseGroup().getExam().isTestExam();
+            return isTestExam ? 2 : 1;
+        }
+        else if (programmingExercise.getDueDate() != null && programmingExercise.getDueDate().isBefore(ZonedDateTime.now())) {
```

Comment:
This should also take the participation with a potentially later individual due date into account